ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2981,8 +2984,15 @@ public Partition createPartition(Table tbl, Map<String, String> partSpec) throws
             : getMSC().add_partitions(partsToAdd, addPartitionDesc.isIfNotExists(), true)) {
           out.add(new Partition(tbl, outPart));
         }
+        EnvironmentContext ec = new EnvironmentContext();
+        // In case of replication statistics is obtained from the source, so do not update those
+        // on replica. Since we are not replicating statistics for transactional tables, do not do
+        // so for a partition of a transactional table right now.
+        if (!AcidUtils.isTransactionalTable(tbl)) {
```

Comment:
This code is in "else" portion of condition
"if (!addPartitionDesc.getReplicationSpec().isInReplicationScope())" a few lines above. So, applies only in case of replication scope. No separate check required.

Reply:
ok